staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -744,40 +757,64 @@ func TestTooWide(t *testing.T) {
 	}.exercise(t)
 }
 
+// TestWindup exercises a scenario with the windup problem.
+// That is, a flow that can not use all the seats that it is allocated
+// for a while.  During that time, the queues that serve that flow
+// advance their `virtualStart` (that is, R(next dispatch in virtual world))
+// more slowly than the other queues (which are using more seats than they
+// are allocated).  The implementation has a hack that addresses part of
+// this imbalance but not all of it.  In this test, flow 1 can not use all
+// of its allocation during the first half, and *can* (and does) use all of
+// its allocation and more during the second half.
+// Thus we expect the fair (not equal) result
+// in the first half and an unfair result in the second half.
+// This func has two test cases, bounding the amount of unfairness
+// in the second half.
 func TestWindup(t *testing.T) {
 	metrics.Register()
-	now := time.Now()
-
-	clk, counter := testeventclock.NewFake(now, 0, nil)
-	qsf := newTestableQueueSetFactory(clk, countingPromiseFactoryFactory(counter))
-	qCfg := fq.QueuingConfig{
-		Name:             "TestWindup",
-		DesiredNumQueues: 9,
-		QueueLengthLimit: 6,
-		HandSize:         1,
-		RequestWaitLimit: 10 * time.Minute,
-	}
-	qsc, err := qsf.BeginConstruction(qCfg, newObserverPair(clk))
-	if err != nil {
-		t.Fatal(err)
+	testCases := []struct {
+		margin2     float64
+		expectFair2 bool
+	}{
+		{margin2: 0.26, expectFair2: true},
+		{margin2: 0.1, expectFair2: false},
```

Comment:
> I am not sure I understand the second point. Is it about the func name, TestWindup? This func is about a scenario that has the windup problem.

It's not about func name - it's about introducing an explicit "name" field in the testcase struct.
This allows a reader to quickly understand what a particular testcase is actually testing, which at least for me - even though I read those tests multiple times, I never remember what margin2 is for example.